google-ads/src/test/functional/java/com/google/ads/googleads/lib/SmokeTest.java

```diff
@@ -0,0 +1,101 @@
+package com.google.ads.googleads.lib;
+
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import static org.junit.Assert.assertTrue;
+
+import com.google.ads.googleads.v7.resources.CustomerName;
+import com.google.ads.googleads.v7.services.CustomerServiceClient;
+import com.google.ads.googleads.v7.services.GoogleAdsServiceClient;
+import com.google.ads.googleads.v7.services.GoogleAdsServiceClient.SearchPagedResponse;
+import com.google.ads.googleads.v7.services.ListAccessibleCustomersRequest;
+import com.google.ads.googleads.v7.services.ListAccessibleCustomersResponse;
+import com.google.api.gax.grpc.GrpcCallContext;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.junit.Test;
+import org.junit.runner.JUnitCore;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.threeten.bp.Duration;
+
+/**
+ * Basic smoke tests for a built jar of the client library.
+ *
+ * <p>Only depends on a basic configuration file with devtoken, OAuth etc.
+ *
+ * <p>Retrieves all accessible customers and iterates through all their campaigns.
+ */
+@RunWith(JUnit4.class)
+public class SmokeTest {
+
+  /**
+   * Provides a main method to run these tests. Useful for testing artifacts such as shadow jar
+   * which require modifying the runtime classpath.
+   */
+  public static void main(String[] args) {
+    System.out.println("Running SmokeTest");
+    JUnitCore.runClasses(SmokeTest.class);
+    System.out.println("Successfully completed SmokeTest");
+  }
+
+  @Test
+  public void ensureCanReadAllAvailableCustomersCampaigns() throws IOException {
+    GoogleAdsClient client = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    List<String> accessibleCustomerIds = getAvailableCustomerIds(client);
+    if (accessibleCustomerIds.isEmpty()) {
+      System.out.println("No accessible customers found. Falling back to nbirnie's $1/day.");
+      accessibleCustomerIds = ImmutableList.of("7731030374");
```

Comment:
+1. You can just add a JUnit assertion here so the test will fail if `accessibleCustomerIds.isEmpty()`.

Reply:
So the problem is that with corp credentials, `CustomerService.listAccessibleCustomers()` returns an empty list. My rationale was that if we get back no accessible customers (but not an authentication exception) then we know that we're running with a corp account. In this case it should be reasonably safe to fallback to someone's $/day. 

I also considered creating a set of credentials and customer ID for running functional tests. However, I decided against this because of the hassle of continually refreshing the refresh tokens (due to Rhea policy).

Reply:
I could also fall back to a generic test account, not specific to any of us, but just something that's got a couple of campaigns?